src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -484,7 +484,7 @@ public override bool CanApplyChange(ApplyChangesKind feature)
             }
         }
 
-        private bool TryGetProjectData(ProjectId projectId, out IVsHierarchy hierarchy, out EnvDTE.Project project)
+        private bool TryGetProjectData(ProjectId projectId, [NotNullWhen(returnValue: true)] out IVsHierarchy? hierarchy, [NotNullWhen(returnValue: true)] out EnvDTE.Project? project)
```

Comment:
>returnValue: true [](start = 73, length = 17)

I'd prefer to not name the argument - remove `returnValue:`. It makes the signature so much longer.

Reply:
Its so much more readable though!

Reply:
IDK, I don't think it adds much value in this case because you already know TryGet pattern is ginna have these.

Reply:
I think it's useful for people who haven't encountered the pattern before. However, good documentation on `NotNullWhen` also solves this problem.